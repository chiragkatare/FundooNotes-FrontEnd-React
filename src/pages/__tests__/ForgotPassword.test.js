import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import ForgotPassword from '../ForgotPassword';

// configure({adapter: new Adapter()});

describe('ForgotPassword', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ForgotPassword debug />);
  
    expect(component).toMatchSnapshot();
  });

  it('email input respond to change event and change state of the Register Component', () => {

    const componentInput = shallow(<ForgotPassword />);
    const emailTest = componentInput.find('Input').at(0);
  
    emailTest.props().onChange({
        target: {
            name: 'email',
            value: 'abcd@gmail.com'
        }
    });
  
    expect(componentInput.state('email')).toEqual('abcd@gmail.com');
  })

  
});