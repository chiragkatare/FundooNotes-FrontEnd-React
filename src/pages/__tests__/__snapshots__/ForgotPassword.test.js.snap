// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword should render correctly in "debug" mode 1`] = `
<div>
  <WithStyles(Card)
    id="logincard"
  >
    <div
      className="label-forgetpass"
    >
      <div
        className="logo"
      >
        <img
          alt=""
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "logo.svg",
            }
          }
        />
      </div>
    </div>
    <WithStyles(CardContent)>
      <div>
        <WithStyles(Typography)
          component="h2"
          id="login-text"
          variant="h5"
        >
          Forgot Password
        </WithStyles(Typography)>
        <div>
          <Input
            label="Email"
            name="email"
            onChange={[Function]}
            placeholder="Enter Your Email"
            type="Email"
          />
        </div>
        <div
          id="login-btn-div"
        >
          <WithStyles(Button)
            className="login-btn"
            color="primary"
            onClick={[Function]}
            type="submit"
            variant="contained"
          >
            Submit
          </WithStyles(Button)>
        </div>
        <span
          id="right-text"
        >
          <a
            href="/register"
          >
            Sign Up
          </a>
        </span>
        <span
          id="left-text"
        >
          <a
            href="/login"
          >
            Login
          </a>
        </span>
      </div>
    </WithStyles(CardContent)>
  </WithStyles(Card)>
</div>
`;
